src/utils/transform.ts

```diff
@@ -111,12 +122,32 @@ export default function transform(
 		if (!plugin.transform) return;
 
 		promise = promise.then(previous => {
+			let assets: Asset[];
 			return Promise.resolve()
 				.then(() => {
-					const context = createPluginTransformContext(graph, plugin, id, previous);
+					let context;
+					({ assets, context } = createPluginTransformContext(
+						graph,
+						plugin,
+						id,
+						previous,
+						createTransformEmitAsset
```

Comment:
An indirection could be avoided if instead of passing on `createTransformEmitAsset`, which is evaluated at `createPluginTransformContext` and just hands us back the assets unchanged, we evaluate `createTransformEmitAsset` here and just pass on the `emitAsset` handler. This would also simplify `createPluginTransformContext` to only return the context.